tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs

```diff
@@ -1,32 +1,493 @@
+
+            var oidcOptions = provider.GetRequiredService<IOptionsFactory<OpenIdConnectOptions>>().Create(_oidcScheme);
+
+            // Assert subscribed to diagnostics
+            if (subscribeToDiagnostics)
+                diagnosticsMock.ReceivedWithAnyArgs().Subscribe(Arg.Any<OpenIdConnectEvents>());
+            else
+                diagnosticsMock.DidNotReceiveWithAnyArgs().Subscribe(Arg.Any<OpenIdConnectEvents>());
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task AddSignIn_WithConfigName_B2cSpecificSetup(bool useServiceCollectionExtension)
+        {
+            var configMock = Substitute.For<IConfiguration>();
+            _configSection = GetConfigSection(_configSectionName, true);
+            configMock.Configure().GetSection(_configSectionName).Returns(_configSection);
+
+            var services = new ServiceCollection();
+            services.AddDataProtection();
+
+            if (useServiceCollectionExtension)
+                services.AddSignIn(configMock, _configSectionName, _oidcScheme, _cookieScheme, false);
+            else
+                new AuthenticationBuilder(services)
+                    .AddSignIn(configMock, _configSectionName, _oidcScheme, _cookieScheme, false);
+
+            await AddSignIn_TestB2cSpecificSetup(services).ConfigureAwait(false);
+        }
+
         [Theory]
-        [InlineData(TestConstants.AuthorityCommonTenant, TestConstants.AuthorityCommonTenantWithV2)]
-        [InlineData(TestConstants.AuthorityOrganizationsUSTenant, TestConstants.AuthorityOrganizationsUSWithV2)]
-        [InlineData(TestConstants.AuthorityCommonTenantWithV2, TestConstants.AuthorityCommonTenantWithV2)]
-        [InlineData(TestConstants.AuthorityCommonTenantWithV2 + "/", TestConstants.AuthorityCommonTenantWithV2)]
-        [InlineData(TestConstants.B2CAuthorityWithV2, TestConstants.B2CAuthorityWithV2)]
-        [InlineData(TestConstants.B2CCustomDomainAuthorityWithV2, TestConstants.B2CCustomDomainAuthorityWithV2)]
-        [InlineData(TestConstants.B2CAuthority, TestConstants.B2CAuthorityWithV2)]
-        [InlineData(TestConstants.B2CCustomDomainAuthority, TestConstants.B2CCustomDomainAuthorityWithV2)]
-        public void EnsureAuthorityIsV2_0(string initialAuthority, string expectedAuthority)
-        {
-            OpenIdConnectOptions options = new OpenIdConnectOptions
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task AddSignIn_WithConfigActions_B2cSpecificSetup(bool useServiceCollectionExtension)
+        {
+            _configureMsOptions = (options) => {
+                options.Instance = TestConstants.AadInstance;
+                options.TenantId = TestConstants.TenantIdAsGuid;
+                options.ClientId = TestConstants.ClientId;
+                options.SignUpSignInPolicyId = TestConstants.B2CSignUpSignInUserFlow;
+                options.Domain = TestConstants.Domain;
```

Comment:
domain should be b2c specific -> `"fabrikamb2c.onmicrosoft.com"`

Reply:
Specific value doesn't affect the outcome of the test but I've updated it for readability.

Reply:
I understand that, but if someone were to read the code in the tests, it doesn't make sense.

---
In reply to: [417818463](https://github.com/AzureAD/microsoft-identity-web/pull/132#discussion_r417818463) [](ancestors = 417818463)